Metzger dynamic inputs

jrmetzger opened the pull request.

aaronlippold commented on kitchen.ec2.yml:
```diff
@@ -0,0 +1,80 @@
+        echo "NOTICE - updating ec2-user sudo config"
+        sudo chmod 600 /etc/sudoers && sudo sed -i'' "/ec2-user/d" /etc/sudoers && sudo chmod 400 /etc/sudoers
+
+        # Ansible Harden did not include - TODO move to main
+        echo "NOTICE - SV-230330: RHEL 8 must not allow users to override SSH environment variables."
+        sudo sed -i 's/.*PermitUserEnvironment.*/PermitUserEnvironment no/g' /etc/ssh/sshd_config
+        echo "NOTICE - SV-230380: RHEL 8 must not allow accounts configured with blank or null passwords"
+        sudo sed -i 's/.*PermitEmptyPasswords.*/PermitEmptyPasswords no/g' /etc/ssh/sshd_config
+        #echo "NOTICE - SV-230251: The RHEL 8 SSH server must be configured to use only Message Authentication Codes (MACs) employing FIPS 140-2 validated cryptographic hash algorithms."
+        # TODO
+        echo "NOTICE - SV-230355: RHEL 8 must map the authenticated identity to the user or group account for PKI-based authentication."
+        sudo echo "[certmap]" > /etc/sssd/sssd.conf
+        sudo mkdir -p /etc/sssd/conf.d
+        echo "NOTICE - SV-244553: RHEL 8 must ignore IPv4 Internet Control Message Protocol (ICMP) redirect messages."
+        sudo sysctl -w net.ipv4.conf.all.accept_redirects=0
+        echo "NOTICE - SV-230325: All RHEL 8 local initialization files must have mode 0740 or less permissive"
+        sudo chmod g-wx,o-rwx $(find /root/.* /home/* -xdev -maxdepth 1 -name '.*' -type f -perm /037)
+        echo "NOTICE - SV-230364: RHEL 8 passwords must have a 24 hours/1 day minimum password lifetime restriction in /etc/shadow."
+        sudo chage -m 1 ec2-user
+        sudo chage -m 1 nobody
+        echo "NOTICE - SV-230496: RHEL 8 must disable the stream control transmission protocol (SCTP)"
+        echo -e "install SCTP /bin/true\nblacklist SCTP" > /etc/modprobe.d/90-sctp
+        echo "NOTICE - SV-230497: RHEL 8 must disable the transparent inter-process communication (TIPC) protocol."
+        echo -e "install TIPC /bin/true\nblacklist TIPC" > /etc/modprobe.d/90-tipc
+        echo "NOTICE -SV-230351: RHEL 8 must be able to initiate directly a session lock for all connection types using smartcard when the smartcard is removed"
+        sudo authselect select sssd with-smartcard with-smartcard-lock-on-removal
+
+        # Requires root?
+        #echo "NOTICE - SV-237643: RHEL 8 must require re-authentication when using the sudo command."
+        #sudo echo 'Defaults timestamp_timeout=0' > /etc/sudoers.d/90-timestamp-timeout
+        #echo "NOTICE - SV-237642: RHEL 8 must use the invoking user's password for privilege escalation when using 'sudo'"
+        #sudo echo 'Defaults !targetpw' > /etc/sudoers.d/90-target-pw
+        #sudo echo 'Defaults !rootpw' > /etc/sudoers.d/90-root-pw
+        #sudo echo 'Defaults !runaspw' > /etc/sudoers.d/90-run-as-pw   
```
Why is all this here? This should be a standard part of the hardening playbook. This needs to be moved to the ansible roles.
```suggestion
        # Ansible Harden did not include - TODO move to main
        echo "NOTICE - SV-230330: RHEL 8 must not allow users to override SSH environment variables."
        sudo sed -i 's/.*PermitUserEnvironment.*/PermitUserEnvironment no/g' /etc/ssh/sshd_config
        echo "NOTICE - SV-230380: RHEL 8 must not allow accounts configured with blank or null passwords"
        sudo sed -i 's/.*PermitEmptyPasswords.*/PermitEmptyPasswords no/g' /etc/ssh/sshd_config
        #echo "NOTICE - SV-230251: The RHEL 8 SSH server must be configured to use only Message Authentication Codes (MACs) employing FIPS 140-2 validated cryptographic hash algorithms."
        # TODO
        echo "NOTICE - SV-230355: RHEL 8 must map the authenticated identity to the user or group account for PKI-based authentication."
        sudo echo "[certmap]" > /etc/sssd/sssd.conf
        sudo mkdir -p /etc/sssd/conf.d
        echo "NOTICE - SV-244553: RHEL 8 must ignore IPv4 Internet Control Message Protocol (ICMP) redirect messages."
        sudo sysctl -w net.ipv4.conf.all.accept_redirects=0
        echo "NOTICE - SV-230325: All RHEL 8 local initialization files must have mode 0740 or less permissive"
        sudo chmod g-wx,o-rwx $(find /root/.* /home/* -xdev -maxdepth 1 -name '.*' -type f -perm /037)
        echo "NOTICE - SV-230364: RHEL 8 passwords must have a 24 hours/1 day minimum password lifetime restriction in /etc/shadow."
        sudo chage -m 1 ec2-user
        sudo chage -m 1 nobody
        echo "NOTICE - SV-230496: RHEL 8 must disable the stream control transmission protocol (SCTP)"
        echo -e "install SCTP /bin/true\nblacklist SCTP" > /etc/modprobe.d/90-sctp
        echo "NOTICE - SV-230497: RHEL 8 must disable the transparent inter-process communication (TIPC) protocol."
        echo -e "install TIPC /bin/true\nblacklist TIPC" > /etc/modprobe.d/90-tipc
        echo "NOTICE -SV-230351: RHEL 8 must be able to initiate directly a session lock for all connection types using smartcard when the smartcard is removed"
        sudo authselect select sssd with-smartcard with-smartcard-lock-on-removal

        # Requires root?
        #echo "NOTICE - SV-237643: RHEL 8 must require re-authentication when using the sudo command."
        #sudo echo 'Defaults timestamp_timeout=0' > /etc/sudoers.d/90-timestamp-timeout
        #echo "NOTICE - SV-237642: RHEL 8 must use the invoking user's password for privilege escalation when using 'sudo'"
        #sudo echo 'Defaults !targetpw' > /etc/sudoers.d/90-target-pw
        #sudo echo 'Defaults !rootpw' > /etc/sudoers.d/90-root-pw
        #sudo echo 'Defaults !runaspw' > /etc/sudoers.d/90-run-as-pw   
```
```suggestion
        # Ansible Harden did not include - TODO move to main
        echo "NOTICE - SV-230330: RHEL 8 must not allow users to override SSH environment variables."
        sudo sed -i 's/.*PermitUserEnvironment.*/PermitUserEnvironment no/g' /etc/ssh/sshd_config
        echo "NOTICE - SV-230380: RHEL 8 must not allow accounts configured with blank or null passwords"
        sudo sed -i 's/.*PermitEmptyPasswords.*/PermitEmptyPasswords no/g' /etc/ssh/sshd_config
        #echo "NOTICE - SV-230251: The RHEL 8 SSH server must be configured to use only Message Authentication Codes (MACs) employing FIPS 140-2 validated cryptographic hash algorithms."
        # TODO
        echo "NOTICE - SV-230355: RHEL 8 must map the authenticated identity to the user or group account for PKI-based authentication."
        sudo echo "[certmap]" > /etc/sssd/sssd.conf
        sudo mkdir -p /etc/sssd/conf.d
        echo "NOTICE - SV-244553: RHEL 8 must ignore IPv4 Internet Control Message Protocol (ICMP) redirect messages."
        sudo sysctl -w net.ipv4.conf.all.accept_redirects=0
        echo "NOTICE - SV-230325: All RHEL 8 local initialization files must have mode 0740 or less permissive"
        sudo chmod g-wx,o-rwx $(find /root/.* /home/* -xdev -maxdepth 1 -name '.*' -type f -perm /037)
        echo "NOTICE - SV-230364: RHEL 8 passwords must have a 24 hours/1 day minimum password lifetime restriction in /etc/shadow."
        sudo chage -m 1 ec2-user
        sudo chage -m 1 nobody
        echo "NOTICE - SV-230496: RHEL 8 must disable the stream control transmission protocol (SCTP)"
        echo -e "install SCTP /bin/true\nblacklist SCTP" > /etc/modprobe.d/90-sctp
        echo "NOTICE - SV-230497: RHEL 8 must disable the transparent inter-process communication (TIPC) protocol."
        echo -e "install TIPC /bin/true\nblacklist TIPC" > /etc/modprobe.d/90-tipc
        echo "NOTICE -SV-230351: RHEL 8 must be able to initiate directly a session lock for all connection types using smartcard when the smartcard is removed"
        sudo authselect select sssd with-smartcard with-smartcard-lock-on-removal

        # Requires root?
        #echo "NOTICE - SV-237643: RHEL 8 must require re-authentication when using the sudo command."
        #sudo echo 'Defaults timestamp_timeout=0' > /etc/sudoers.d/90-timestamp-timeout
        #echo "NOTICE - SV-237642: RHEL 8 must use the invoking user's password for privilege escalation when using 'sudo'"
        #sudo echo 'Defaults !targetpw' > /etc/sudoers.d/90-target-pw
        #sudo echo 'Defaults !rootpw' > /etc/sudoers.d/90-root-pw
        #sudo echo 'Defaults !runaspw' > /etc/sudoers.d/90-run-as-pw   
```
```suggestion
        # Ansible Harden did not include - TODO move to main
        echo "NOTICE - SV-230330: RHEL 8 must not allow users to override SSH environment variables."
        sudo sed -i 's/.*PermitUserEnvironment.*/PermitUserEnvironment no/g' /etc/ssh/sshd_config
        echo "NOTICE - SV-230380: RHEL 8 must not allow accounts configured with blank or null passwords"
        sudo sed -i 's/.*PermitEmptyPasswords.*/PermitEmptyPasswords no/g' /etc/ssh/sshd_config
        #echo "NOTICE - SV-230251: The RHEL 8 SSH server must be configured to use only Message Authentication Codes (MACs) employing FIPS 140-2 validated cryptographic hash algorithms."
        # TODO
        echo "NOTICE - SV-230355: RHEL 8 must map the authenticated identity to the user or group account for PKI-based authentication."
        sudo echo "[certmap]" > /etc/sssd/sssd.conf
        sudo mkdir -p /etc/sssd/conf.d
        echo "NOTICE - SV-244553: RHEL 8 must ignore IPv4 Internet Control Message Protocol (ICMP) redirect messages."
        sudo sysctl -w net.ipv4.conf.all.accept_redirects=0
        echo "NOTICE - SV-230325: All RHEL 8 local initialization files must have mode 0740 or less permissive"
        sudo chmod g-wx,o-rwx $(find /root/.* /home/* -xdev -maxdepth 1 -name '.*' -type f -perm /037)
        echo "NOTICE - SV-230364: RHEL 8 passwords must have a 24 hours/1 day minimum password lifetime restriction in /etc/shadow."
        sudo chage -m 1 ec2-user
        sudo chage -m 1 nobody
        echo "NOTICE - SV-230496: RHEL 8 must disable the stream control transmission protocol (SCTP)"
        echo -e "install SCTP /bin/true\nblacklist SCTP" > /etc/modprobe.d/90-sctp
        echo "NOTICE - SV-230497: RHEL 8 must disable the transparent inter-process communication (TIPC) protocol."
        echo -e "install TIPC /bin/true\nblacklist TIPC" > /etc/modprobe.d/90-tipc
        echo "NOTICE -SV-230351: RHEL 8 must be able to initiate directly a session lock for all connection types using smartcard when the smartcard is removed"
        sudo authselect select sssd with-smartcard with-smartcard-lock-on-removal

        # Requires root?
        #echo "NOTICE - SV-237643: RHEL 8 must require re-authentication when using the sudo command."
        #sudo echo 'Defaults timestamp_timeout=0' > /etc/sudoers.d/90-timestamp-timeout
        #echo "NOTICE - SV-237642: RHEL 8 must use the invoking user's password for privilege escalation when using 'sudo'"
        #sudo echo 'Defaults !targetpw' > /etc/sudoers.d/90-target-pw
        #sudo echo 'Defaults !rootpw' > /etc/sudoers.d/90-root-pw
        #sudo echo 'Defaults !runaspw' > /etc/sudoers.d/90-run-as-pw   
```

aaronlippold commented on controls/SV-230274.rb:
```diff
@@ -14,71 +14,56 @@
     unless sssd_conf_file_contents.empty?
-      describe ini({ command: 'cat /etc/sssd/sssd.conf /etc/sssd/conf.d/*.conf'}) do
-        its('sssd.certificate_verification') { should cmp 'ocsp_dgst=sha1' }
+      describe ini({ command: 'cat /etc/sssd/sssd.conf /etc/sssd/conf.d/*.conf' }) do
```
This does not seem quite right to me

aaronlippold commented on controls/SV-244546.rb:
```diff
@@ -18,114 +20,86 @@
+    if file(rules_file).exist?
+      describe file(rules_file).content.strip.split("\n")[-1] do
+        it { should cmp 'deny perm=any all : all' }
+      end
```
@wdower @em-c-rod we should verify that this failure is not perhaps a typo in the guidance.
```suggestion
Verify this is not a typo in the guidance, given this is how the Ansible Lockdown hardening configures it this way for v1r12
```ruby
  expected: deny perm=any all : all
  got: deny all all```
```

aaronlippold commented on controls/SV-230229.rb:
```diff
@@ -54,33 +51,31 @@
+  end
+  describe x509_certificate(root_ca_file) do
     its('issuer_cn') { should match 'CN=DoD' }
   end
```
@wdower @em-c-rod this needs to be added to the x509 certificate resource as a properity that so we can pass in the 'child' and 'parent' files and get back a bool to indicate a 'cert is part of a chain'

wdower committed:
using include instead of match on 255924

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

aaronlippold committed:
small things and formatting

Signed-off-by: Aaron Lippold <lippold@gmail.com>

jrmetzger committed:
Merge branch 'metzger_dynamic_inputs' of https://github.com/mitre/redhat-enterprise-linux-8-stig-baseline into metzger_dynamic_inputs

jrmetzger committed:
gpg keys check

jrmetzger committed:
rpm gpg file/keys

jrmetzger committed:
rpm gpg keys inspec update

jrmetzger committed:
rpm gpg keys/file

wdower committed:
adding wip for 254520

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

jrmetzger committed:
fix_logind_conf

jrmetzger committed:
Merge branch 'metzger_dynamic_inputs' of https://github.com/mitre/redhat-enterprise-linux-8-stig-baseline into metzger_dynamic_inputs

aaronlippold committed:
removed broken control until we fix

Signed-off-by: Aaron Lippold <lippold@gmail.com>

wdower committed:
fixing weird inputs bug

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
fixing 254520

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

aaronlippold committed:
Updates:

- SV-230251: Reviewed, updated and improved implementation, added input
  and clarified communition on both vanilla and hardened testing and
  updated control-status table accordeningly
- Updated the vanilla and hardened threshold to hold us to a more
  accurate bseline
- Removed unused control data
- Added the scan results used to create the new vanilla and hardened
  threshold into the spec directory

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
testing the t3a instance type

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
testsing m5.large

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added continue-on-error to workflow for Heimdall Upload

Signed-off-by: Aaron Lippold <lippold@gmail.com>

wdower committed:
filling out control inputs in the control status table

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
adding an input to 257258

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updating 230233

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updated 230319 to use expect syntax

Signed-off-by: wdower <will@dower.dev>

wdower committed:
updating 230243 to use expect syntax

Signed-off-by: wdower <will@dower.dev>

wdower committed:
updating threshold for vanilla, fixing typo in 230243

Signed-off-by: wdower <will@dower.dev>

wdower committed:
updating hardened threshold to accurately reflect expected failure in 230233

Signed-off-by: wdower <will@dower.dev>

jrmetzger committed:
sha typo

jrmetzger committed:
fix typo

jrmetzger committed:
broke up inputs

wdower committed:
removing service input from 251710 since aide and other fim tools arent services

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
using cmp instead of be in 251716

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updated vanilla threshold to match expected output

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
removing firewalld since it is an official inspec resource now

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

aaronlippold commented on controls/SV-230251.rb:
```diff
@@ -1,83 +1,75 @@
+  config_options.transform_values! { |v| v.split(",") }
+
+  # Define the path to the crypto policy file
+  crypto_policy_file = '/etc/crypto-policies/back-ends/opensshserver.config'
```
testing comment @em-c-rod

aaronlippold commented on controls/SV-230251.rb:
```diff
@@ -1,83 +1,75 @@
+  required_algorithms = input('openssh_server_required_algorithms')
+
+  # TODO: make a simple resource for this based off 'login_defs' or 'yum' as a model
+
```
@em-c-rod   # make a simple resource for this based off 'login_defs' or 'yum' as a model

aaronlippold commented on controls/SV-230251.rb:
```diff
@@ -1,83 +1,75 @@
+  required_algorithms = input('openssh_server_required_algorithms')
+
+  # TODO: make a simple resource for this based off 'login_defs' or 'yum' as a model
+
+  # Parse the configuration file to get the value of "CRYPTO_POLICY"
+  crypto_policy = parse_config_file("/etc/crypto-policies/back-ends/opensshserver.config")["CRYPTO_POLICY"]
+
+  # Parse the CRYPTO_POLICY string into a hash of configuration options
+  config_options = crypto_policy.scan(/-o(\w+)=([\w\-,@]+)/).to_h
+
+  # Split each configuration option's values into an array
+  config_options.transform_values! { |v| v.split(",") }
+
```
@em-c-rod this is the basis of small resource

aaronlippold committed:
Organized and put things in closets

- Organized and clearly marked things so folks knew easily what was what

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added do-not-merge back for now

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
only-validate-profile

Removed unessary files and looking to see if I can make the workflow
runs a bit more speedy with some keywords

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
trying again, only-validate-profile

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
Reviewed and tested SV-230244 and added container testing

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
moved platform definition down to the sub kitchnen files to make the platform names align with the testing target type

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
to err is human, to find the bug, devine... this really better pass now :)

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
adding missing env vars for the container images and moving linting to a seperate process so we only do it once per workfow

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
updating the name of the new linting task

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
made the verify of ec2 and ubi8 dependant on the lint and check

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
linting doesn't need to be a matrix

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
trying triggered workflows again

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
fixing syntax on lint workflow

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added documentation for profile update process, and incorporated the inspec delta rubocop rules into the profiles .rubocop file

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added missing sections to the docs and refined the delta workflow instructions

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
rounding out documentation sections

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added a TOC to the markdown file

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added section numbers and updated TOC

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
fixing some broken internal links

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added sections explaining the hardening, roles, use of Ansible Lockdown etc.

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added pdf version

Signed-off-by: Aaron Lippold <lippold@gmail.com>

em-c-rod committed:
update SV-230222 to have metadata from delta run

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

aaronlippold committed:
removed the do-not-merge

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
rubocop, we must fix this later:

Signed-off-by: Aaron Lippold <lippold@gmail.com>

jrmetzger committed:
fix input sssd_certificate_verification

jrmetzger committed:
Merge branch 'metzger_dynamic_inputs' of https://github.com/mitre/redhat-enterprise-linux-8-stig-baseline into metzger_dynamic_inputs

jrmetzger committed:
exit status for aide

em-c-rod committed:
Merge branch 'main' of https://github.com/mitre/redhat-enterprise-linux-8-stig-baseline into metzger_dynamic_inputs

em-c-rod committed:
Merge branch 'metzger_dynamic_inputs' of https://github.com/mitre/redhat-enterprise-linux-8-stig-baseline into metzger_dynamic_inputs

em-c-rod committed:
update controls for consistent formatting with main and delta

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
remove os_versions input and change the logic to use only_if statements. Also, remove comment for unnecessary check in 251710 and adjust hardened threshold accordingly

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

aaronlippold committed:
reformatted control status table

Signed-off-by: Aaron Lippold <lippold@gmail.com>

em-c-rod committed:
update SV-230233 to reflect the updates in main

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
update hardened and vanilla thresholds for updated SV-230233 control

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
delete duplicate difok entry in inspec.yml and add input usage back to SV-230233

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
update messages that allow not applicable controls from inputs and fix a typo

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

aaronlippold committed:
fixed broken inputs that were not wrapped in quotes like they should have been

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
I think the update wording is a bit more clear

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
moving thredholds back until we have a more structured process of updating them for the pipeline

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
moving updating and development of profile work to training site, added details in readme respectivily

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added actual linting to the profile, add bundle exec rake inspec:check and pre_commit_checks

Signed-off-by: Aaron Lippold <lippold@gmail.com>

em-c-rod committed:
update some skip statements, fix control with two only_ifs, remove unused input in inspec.yml, remove some comments and redundancies

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
removing the check the was moved to SV-251718

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
fix typo of elsif to else

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

wdower committed:
linting

Signed-off-by: wdower <will@dower.dev>

em-c-rod committed:
update SV-230244 to simplify and correct not applicability

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
pin saf_action to v1.5.0 after release errors

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
rename smart_card_status to smart_card_enabled, update saf_action to v1.5.0, update control SV-230372

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
updated some docs and testing latest release of SAF Action

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
pin SAF action back to 1.5.0

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

wdower committed:
updating control-status to correctly indicate that we already updated all metadata to v1r12 with saf generate delta

Signed-off-by: wdower <will@dower.dev>

wdower committed:
fixing 230233

Signed-off-by: wdower <will@dower.dev>

wdower committed:
removing ghost oldControls

Signed-off-by: wdower <will@dower.dev>

em-c-rod committed:
changing so that disable_slow_controls causes a not review instead of a not applicable status for SV-230222

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

aaronlippold commented on controls/SV-250317.rb:
```diff
@@ -0,0 +1,130 @@
+      end
+    end
+
+    # Get the list of sysctl configuration files
+    sysctl_config_files = input('sysctl_conf_files').map(&:strip).join(' ')
+
+    # Search for the kernel parameter in the configuration files
+    search_results = command("grep -r #{parameter} #{sysctl_config_files} {} \;").stdout.split("\n")
+
+    # Parse the search results into a hash
+    config_values = search_results.each_with_object({}) do |item, results|
+      file, setting = item.split(':')
+      results[file] ||= []
+      results[file] << setting.split('=').last
+    end
```
This should really be pulled out into a simple generalized resource

aaronlippold replied:
@em-c-rod @wdower

aaronlippold replied:
Lines 98 to 109 ish I mean should be a resource we can reuse across these types of checks

wdower committed:
adding assignees to each control (initial)

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

em-c-rod committed:
I can take SV-230229 since I was just looking at it - Emily

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

jrmetzger committed:
param instead of bracket regex

aaronlippold committed:
updated gem file

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
updated gem file

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
fixed conflict in table

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
adding back table since i wnet the wrong way, also fixing inputs, the <br> go OUTSIDE the  marks.

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
trying to fix running of actions

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold commented on controls/SV-245540.rb:
```diff
@@ -35,19 +32,24 @@
     end
   else
-    describe package('mcafeetp') do
+    linux_threat_prevention_package = input('linux_threat_prevention_package')
```
I would guess we really should add an input here to 'check_via_service' and 'check_via_process' since if they are using another tool it may use a service vs a process approach.

aaronlippold replied:
This input should be renamed to process since that is what we are checking

aaronlippold commented on controls/SV-244554.rb:
```diff
@@ -1,56 +1,67 @@
   else
     describe kernel_parameter('net.core.bpf_jit_harden') do
       its('value') { should eq 2 }
     end
```
This part of the code should really be a resource

aaronlippold commented on controls/SV-244552.rb:
```diff
@@ -1,56 +1,68 @@
   tag cci: ['CCI-000366']
   tag nist: ['CM-6 b']
 
   if virtualization.system.eql?('docker')
```
Check the logic here, with the statement at the end about only applying to incoming traffic when acting as a router.

aaronlippold commented on controls/SV-244551.rb:
```diff
@@ -1,56 +1,68 @@
   tag cci: ['CCI-000366']
   tag nist: ['CM-6 b']
 
   if virtualization.system.eql?('docker')
```
Check the logic here as to when this applies given the statment about being a router.

aaronlippold commented on controls/SV-250317.rb:
```diff
@@ -0,0 +1,130 @@
+  tag cci: ['CCI-000366']
+  tag nist: ['CM-6 b']
+
+  only_if('This system is acting as a router on the network, this control is Not Applicable', impact: 0.0) {
+    !input('network_router')
+  }
```
Check the logic here as it acting as a router, do we have this reversed?

wdower replied:
This is the right pattern. Control only applies if the network_router input is NOT set

aaronlippold commented on controls/SV-244553.rb:
```diff
@@ -1,56 +1,69 @@
-    Verify RHEL 8 ignores IPv4 ICMP redirect messages.
+  title 'RHEL 8 must ignore IPv4 Internet Control Message Protocol (ICMP)
+redirect messages.'
+  desc "ICMP redirect messages are used by routers to inform hosts that a more direct route exists for a particular destination. These messages modify the host's route table and are unauthenticated. An illicit ICMP redirect message could result in a man-in-the-middle attack.
```
It seems odd to me that this requirement doesn't have the same router caveat as the other simular controls, this may be a tech writter error. Let's see if we can verify that this should be in the same class from older guidance documents.

aaronlippold commented on controls/SV-244554.rb:
```diff
@@ -1,56 +1,67 @@
-  "
+  title 'RHEL 8 must enable hardening for the Berkeley Packet Filter
+Just-in-time compiler.'
+  desc 'It is detrimental for operating systems to provide, or install by default, functionality exceeding requirements or mission objectives. These unnecessary capabilities or services are often overlooked and therefore may remain unsecured. They increase the risk to the platform by providing additional attack vectors.
```
Let's also verify if this requirement should also have the same router language and ncheck as the other controls in this class.

aaronlippold commented on controls/SV-244549.rb:
```diff
@@ -15,46 +15,36 @@
     describe package('openssh-server') do
       it { should be_installed }
     end
   end
```
This should really be amended to only be NA if we are in a container and openssh-server is not installed.

aaronlippold committed:
Control Updates

- Updated implementation of kernel paramater tests to better capture the
  intent of the test
- Enhanced the openssh-server validation to allow for an 'exception' in
  container use of openssh as we have seen this in the field

Signed-off-by: Aaron Lippold <lippold@gmail.com>

wdower committed:
adding CHEF_LICENSE envar to kitchen yml

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updating 23022{5,6}

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
typo in control-status was bugging me

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updating 230227, updating all banner message controls to be consistent with each other and better output

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
linting

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
cleaning up and adding only_if to 230228

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

aaronlippold committed:
added better ec2 labeling and update SV-244549

Signed-off-by: Aaron Lippold <lippold@gmail.com>

wdower committed:
updating 230230 for cleaner logic and better corner case checking

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
linting, again

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updated 230232

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
minor tweaks to 23023{4-7}

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

aaronlippold committed:
SV-244548, added inputs, improved languge, only_if

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
SV-244547, same treatment as ...548

Signed-off-by: Aaron Lippold <lippold@gmail.com>

wdower committed:
updating 2302{38-41}

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
alas, lint

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

aaronlippold commented on controls/SV-244540.rb:
```diff
@@ -28,8 +23,9 @@
   tag nist: ['CM-6 b']
 
-  describe command('grep -i nullok /etc/pam.d/system-auth') do
+  pam_auth_files = input('pam_auth_files')
```
Since we unify the checks of all the pam files for this same setting this control, 244541 and their brother that checks smart cards will all be the same test.

aaronlippold commented on controls/SV-244538.rb:
```diff
@@ -81,20 +72,17 @@
+    impact 0.0
+    describe 'The GNOME desktop is not installed' do
+      skip 'The GNOME desktop is not installed, this control is Not Applicable.'
     end
```
This test set needs enhancement since other destkops are now becoming popular in the DOD. I updated the 'GNOME desktop check to see if any GUI desktop is installed in the xsystem but now we need to add the checks for other supported destkops'

aaronlippold commented on controls/SV-244535.rb:
```diff
@@ -65,21 +56,18 @@
+      skip 'Control not applicable within a container'
+    end
+  elsif package('gnome-desktop3').installed?
+    describe command('gsettings get org.gnome.desktop.screensaver lock-delay') do
```
test enhancements to support other GUI desktops

aaronlippold commented on controls/SV-244534.rb:
```diff
@@ -63,17 +55,14 @@ module in the /etc/pam.d/password-auth file.
+    RHEL version 8.0 or 8.1, this check is not applicable.', impact: 0.0) {
+    (os.release.to_f) >= 8.2
+  }
+
```
verify if this control should also be using the 'pam_auth_files' input

aaronlippold commented on controls/SV-244531.rb:
```diff
@@ -39,16 +34,14 @@
   end
   describe findings do
     it { should be_empty }
   end
```
update this control to use the 'more_permissive_than' matcher - I spent a lot of time writting that :)

aaronlippold commented on controls/SV-244531.rb:
```diff
@@ -39,16 +34,14 @@
   tag cci: ['CCI-000366']
   tag nist: ['CM-6 b']
 
```
in the rhel7 profile this is NA for the container, we should verify if that should be the case here

wdower replied:
I'd argue that a container shouldn't have interactive shells in the first place. Nobody should be shelling into them in prod.

aaronlippold commented on controls/SV-244532.rb:
```diff
@@ -51,16 +46,14 @@
   tag cci: ['CCI-000366']
   tag nist: ['CM-6 b']
 
```
in RHEL7 this is NA for the container, perhaps that should be the case here

aaronlippold commented on controls/SV-244529.rb:
```diff
@@ -1,41 +1,35 @@
     describe etc_fstab.where { mount_point == '/var/tmp' } do
       it { should exist }
     end
   end
```
I would love to verify if the CSP / AMI world may have a differnt read on this, and perhaps its NA in those enviroments

aaronlippold commented on controls/SV-244525.rb:
```diff
@@ -1,83 +1,51 @@
     describe sshd_config do
       its('ClientAliveInterval') { should cmp <= '600' }
     end
   end
```
hardcoded data value, should be an input

aaronlippold commented on controls/SV-244528.rb:
```diff
@@ -1,53 +1,46 @@
     describe sshd_config do
       its('GSSAPIAuthentication') { should cmp 'no' }
     end
   end
```
hard coded data value, should be an input

aaronlippold replied:
can I get a double check on the logic here

aaronlippold committed:
made updates to many controls, added new inputs, updated logic and coverage

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold commented on controls/SV-237643.rb:
```diff
@@ -1,60 +1,53 @@
     describe config do
-      its('stdout') { should match /^Defaults timestamp_timeout/ }
+      its('stdout') { should match(/^Defaults timestamp_timeout/) }
     end
```
convert this into a real and why don't we have it yet - resource for sudoers ....

aaronlippold committed:
made updates to a few more, major refactor on a few, discovered two more new resources we need that are currently in controls, all is well

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
cleaned up the status table

Signed-off-by: Aaron Lippold <lippold@gmail.com>

wdower committed:
tagging some controls for container applicability

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updates through 230254

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
I know what you're thinking -- 'did he lint before the last commit?' The answer is no

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

em-c-rod committed:
updating SV-230229's quotation and spacing, clarity using expect syntax, adding smart_card_enabled check, not tested locally

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
updated not applicable to have impact 0.0 for SV-230229 and SV-230372

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

aaronlippold commented on controls/SV-230229.rb:
```diff
@@ -23,64 +23,74 @@
+  root_ca_file = input('root_ca_file')
+  root_ca_full_path = File.join(root_ca_loc, root_ca_file)
 
+  describe directory(root_ca_loc) do
+    it { should exist }
+  end
+
+  describe file(root_ca_full_path) do
+    it { should exist }
+  end
```
These kinds of tests are not really necessary and if we need to check if a file exists to protect aginst a profile error it should be done in the code logic not as a test.

aaronlippold commented on controls/SV-230372.rb:
```diff
@@ -40,57 +35,71 @@
   tag fix_id: 'F-33016r567863_fix'
   tag cci: ['CCI-000765']
   tag nist: ['IA-2 (1)']
 
```
missing tags for 'host' and 'container'

wdower committed:
23025{5,6}

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
linting

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updating 2302{57-62}

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

em-c-rod committed:
update SV-230229 to consolodate to one input, add host tag to SV-230372

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

aaronlippold committed:
Added DISA Ansible Supplamental Context for diff

- added disa 'harden only' ktichen and added it to the workflow
- using micro box

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
deconflicted file names and added 'disa-'

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added tag to the disa upload to add the version and source of the hardeneing content

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold commented on kitchen.ec2.yml:
```diff
@@ -0,0 +1,62 @@
+  pre_converge:
+    - remote: |
+        echo "NOTICE - Installing needed packages"
+        sudo dnf -y clean all
+        sudo dnf -y install --nogpgcheck bc bind-utils redhat-lsb-core vim
+        echo "updating system packages"
+        sudo dnf -y update --nogpgcheck --nobest
+        sudo dnf -y distro-sync
```
The order and redudent update commands should be cleaned up here

aaronlippold committed:
Fixes for Ansible

- updated it so ansible is installed via dnf in pre-converge
- readded 'max_ssh_sessions' which seems to have gotten lost to fix
  upload error

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
fixing ansible version command since it didn't seem to like '-v'

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
Fixed:

- fixed the ansible version in the base ec2 kitchen file
- added 'train-awsssm' to the Gemfile for general use and avaliblity for
  end-users

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
Changes:

- added disa 63% threshold
- think I have the ansible config and collections working now

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added github action job summary from compliance summary, better managment of ec2 terminiation'

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
ok. stop being fancy

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
trying the summary again

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
...

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
adjusted vanilla threshold

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added GitHub Summary to other pipelines

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
container workflow for GitHub Summary should be fixed and generalized the action workflow to use a PLATFORM env var to avoid simular issue in the future

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
added the same generalization to the other two workflows

Signed-off-by: Aaron Lippold <lippold@gmail.com>

wdower committed:
adding 23026{3-6}

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
realized there's already an input for sysctl config files

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updating 23026{6-8} for consistency and clarity

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
adding 2302{60,70}, adding '^' to grep commands for the kernel controls to make sure we don't catch commented lines

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
adding 23027{0,1}

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

aaronlippold committed:
removed svgs that were part of old docs

Signed-off-by: Aaron Lippold <lippold@gmail.com>

wdower committed:
wip for a sudoers resource borrowing code from 237643 -- still needs a filtertable for user permissions

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
adding rudimentary sudeors resource

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
fixed bug in sudoers to let it handle multiple sudo files

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
fixing bug in settings hash to allow it to parse settings with any amount of whitespace in between key and value

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
removing accidentally added bogus sudoers files I was using for testing

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
adjusting sudoers to collect sudoers files contents by catting them instead of using the file resource so that we can pass a glob to it, made the rules table avoid setting the optional fields to nil

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
fixing busted logic for sudoers to convert a string param to an array

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updating sudoers tests to use local resource

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

aaronlippold committed:
cleanup and removed extra file

Signed-off-by: Aaron Lippold <lippold@gmail.com>

wdower committed:
updating 230271 to use sudeors resource

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
linting

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updated 230274 -- added an input for alternate MFA options to match check text, updated logic for clarity and sanity

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
forgot an input in control-status

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
fixing the tag opn 230274

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 230276

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updated 230277

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
catching nil case in 230274

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
fixing 230274, correctly this time, to catch case of empty sssd conf files, also added an input for what those files are

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
accidentally left in wrong test value from fuzz testing this control

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
standardizing the GRUB setting controls

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 230283

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updates through 230299

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
how much lint could a linter lint if a linter could lint lint?

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
another linter tweak

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
first pass at refactoring 23030{0-2}

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
lint

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
uncommenting 301

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
adding 230303

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
linterino

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
rounding out the etc_fstab stuff

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
better implementation for 301, more tests through 310

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
230211

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
added caveats to 23031{0,1}

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
did a bunch of the world-writeable controls. even remembered to lint before pushing this time

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updated 230313, added caveat input

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 230322

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
got the last homedir check

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

jrmetzger committed:
Merge branch 'main' into metzger_dynamic_inputs

wdower committed:
deciding that reqs related to interactive users would be inapplicable to containers

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 230331, linted

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
fixing incorrect version check in 230334

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 230336

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 230345 on PAM rules -- also refactored 230374 a little bit to match the other temp account req in 230331; it was previously erroring out the pipeline because it didn't have a catch for temp accounts that didnt actually exist

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
lint

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updated limits.conf tests

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
A lint, a lint, my kingdom for a lint!

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
just realized we had a typo in dozens of the container caveats

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
tmux controls

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 230355

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
added 230356 and added a bunch of tags to controls that didn't have them yet

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 230384

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 230403 -- mostly adding a few inputs and updating container check logic to use only_if

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

jrmetzger committed:
fixed present? error, updated

jrmetzger committed:
fix seperate_filesystem_exempt value syntax

jrmetzger committed:
added input unapproved_ssl_tls_versions

jrmetzger committed:
updated .present? and sudoers resource to count for nil

wdower committed:
audit reqs -- up through 230467

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

jrmetzger committed:
fixed audit_command typo

jrmetzger committed:
fixed incorrect controls

jrmetzger committed:
updated controls

wdower committed:
up through 230475

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
fixing accidentally reversed nilcheck in 230353

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
linting

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 484

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
better skip message for alt logging controls

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

jrmetzger committed:
audit_rule_keys input

jrmetzger committed:
fix cookstyle

jrmetzger committed:
update syntax for actual failures

jrmetzger committed:
audit tools group owned by root

jrmetzger committed:
reject good files

jrmetzger committed:
fix broken kernel_config_files input

wdower committed:
up through 230504

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
liiiiiiiiiiiiiiint

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 230522

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

jrmetzger committed:
ignore if nil repos

wdower committed:
up through 230554, some minor tweaks to the new structure for ipv-whatever controls for consistency and clarity

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
missed a few kernel param ones

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updating workflows to check and see if new AWS creds work as expected

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 230558

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
reverting to old AWS credentials while we work out the new env

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
up through 237642

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
adding explicit existence check for some of the file config tests

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updated 230317 for clarity and for some edge cases it did not catch originally

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
forgot to wrap an input in 'input()'

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
linting 317

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
adjusting the thresholds to match expected state in the pipeline

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

em-c-rod committed:
update SV-230233 to put it back again for allowing nil to pass, update Sv-230390 and SV-230392 to change value to uppercase so we don't fail on lower case

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
update to max instead of min copy pasta, but need to resolve profile error

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

wdower committed:
lint

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
fixing strange input bug in 230233

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
explicitly upcasing both sides of the comparison for 230390

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updating 230392

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updating readme and removing wonky stuff in inspec.yml

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

em-c-rod committed:
add a dot to the regex options on opensshserver config for SV-230251

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
lint SV-230317 and SV-230392

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
reset SV-230317 lint and disable the lint rule for this case

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
remove full list of mounted removable media in the error message for easier understanding

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
remove non required pam checks as was done in PR 27 for SV-230356

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

em-c-rod committed:
update use of audit_rule_keys input in each control looking for identity

Signed-off-by: Emily Rodriguez <ecrodriguez@mitre.org>

wdower committed:
updating audit controls to use an input for keynames

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updating README, fixing lint error in 230317

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
adding a second input to the audit controls specifically for keyname overrides, to make sure people can edit a subset of the keynames without throwing out all the rest of the defaults

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
making 230256 an array input

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
cleanup -- removing extraneous files, fixiing tmp mount checks to have input caveats

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
flipping the sign on the tmp mount ones

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
re-adding matchers now that I know what it does (requirement for pam rules)

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
removing unused input for audit keynames

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>

wdower committed:
updating README to describe attestations

Signed-off-by: wdower <57142072+wdower@users.noreply.github.com>
